doc/source/whatsnew/v1.1.0.rst

```diff
@@ -223,6 +223,8 @@ Other
 - Appending a dictionary to a :class:`DataFrame` without passing ``ignore_index=True`` will raise ``TypeError: Can only append a dict if ignore_index=True``
   instead of ``TypeError: Can only append a Series if ignore_index=True or if the Series has a name`` (:issue:`30871`)
 - Set operations on an object-dtype :class:`Index` now always return object-dtype results (:issue:`31401`)
+- Bug in :meth:`AbstractHolidayCalendar.holidays` when no rules were defined (:issue:`31415`)
```

Comment:
We ought to limit notes in the Other section if at all possible if someone wants to try to classify these better

cc @pandas-dev/pandas-core